extensions/sql-migration/src/dialog/skuRecommendationResults/GenerateProvisioningScriptDialog.ts

```diff
@@ -100,17 +100,18 @@ export class GenerateProvisioningScriptDialog {
 		saveTemplateButton.onDidClick(async () => {
 			const folder = await utils.promptUserForFolder();
 			if (folder) {
-				if (this._armTemplateText) {
-					let templateName = utils.generateTemplatePath(this.model, this._targetType);
-					let destinationFilePath = path.join(folder, templateName);
-					try {
-						fs.writeFileSync(destinationFilePath!, this._armTemplateText);
-						void vscode.window.showInformationMessage(constants.SAVE_TEMPLATE_SUCCESS);
-					}
-					catch (e) {
-						logError(TelemetryViews.ProvisioningScriptWizard, 'ArmTemplateSavetoLocalError', e);
-						void vscode.window.showErrorMessage(constants.SAVE_TEMPLATE_FAIL);
+				const templates = this.model._armTemplateResult.templates!;
+				try {
+					for (let i = 0; i < templates.length; i++) {
```

Comment:
batch should not be in the name of MI and VM template file name also if for DB we have just 1 template it should not be there, it should come only if we have more than 1 DB template. 

Maybe you can have a check on number of Templates.

Reply:
Yes, we are handling the names this way only.